compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -260,6 +260,13 @@ impl<K: DepKind> DepGraph<K> {
 
             let print_status = cfg!(debug_assertions) && cx.debug_dep_tasks();
 
+            if matches!(dep_node_index, None) {
+                return (result, self.next_virtual_depnode_index());
+            }
+
+            // Handled None case above, safe to unwrap here.
+            let dep_node_index = dep_node_index.unwrap();
```

Comment:
You can use
``rust
let dep_node_index = if let Some(dep_node_index) = dep_node_index {
    dep_node_index
} else {
    return ...
};
```

Reply:
fair, I'll update once the perf run goes through